subsys/spm/CMakeLists.txt

```diff
@@ -5,3 +5,30 @@
 #
 
 zephyr_sources(spm.c)
+
+if (IMAGE)
+  # Only perform merge operation when being part of multi image build.
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE
+    ${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
+
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE_TARGET
+    ${IMAGE}kernel_elf
+    )
+
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE
+    ${CMAKE_BINARY_DIR}/zephyr/${KERNEL_HEX_NAME}
+
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    HEX_FILES_TO_MERGE_TARGET
+    kernel_elf
```

Comment:
Should this be ${logical_target_for_zephyr_elf} ?

Reply:
That does not exist in this namespace. Will see if I can get its value somehow

Reply:
To be able to use this value I must run these commands 

```
# Read global variables into local variables
get_property(GKOF GLOBAL PROPERTY ${IMAGE}GENERATED_KERNEL_OBJECT_FILES)
get_property(GKSF GLOBAL PROPERTY ${IMAGE}GENERATED_KERNEL_SOURCE_FILES)

``` 

Which looks a bit worse IMO

Reply:
Added to TODO list.